### PR TITLE
donkeycar: initial Dockerfile build

### PR DESCRIPTION
**Pull request by jimcadden**
Donkeycar: updated default model & config
Donkeycar: add initial Dockerfile 

Signed-off-by: Jim Cadden <jcadden@ibm.com>

### CODE REVIEW

File: .aicoe-ci.yaml [hunk 1]
```diff
@@ -5,10 +5,10 @@ check:
   - thoth-build
 # Uncomment following lines to build a public image of this repo
 build:
-  base-image: "quay.io/thoth-station/s2i-elyra-custom-notebook:v0.4.1"
-  build-source-script: "image:///opt/app-root/builder"
+  build-stratergy: Dockerfile # Allowed values: Source, Dockerfile, Containerfile (default: Source)
```
Comment (MichaelClifford) [inline .aicoe-ci.yaml:8]: I think this change will break our ability to build a meteor jupyter lab working environment. @harshad16 am I misyaken?

Comment (harshad16) [reply .aicoe-ci.yaml:8]: Definitely, this would make the pipeline build the image based on Dockerfile.

File: .aicoe-ci.yaml [hunk 2]
```diff
@@ -5,10 +5,10 @@ check:
...
+  base-image: haoru233/tritonai-donkeycar:4.1-cpu # Temporary base img, will replace with our own
```
Comment (harshad16) [inline .aicoe-ci.yaml:9]: if we are building based on the dockerfile, we can remove the base-image

**Review by harshad16 — COMMENTED** [review]
we should discuss on what this repo is for 
* For Donkeycar build
or 
* For notebook build

File: .aicoe-ci.yaml (hunk 1 above)
Comment (jimcadden) [reply .aicoe-ci.yaml:8]: Ok, that's not going to work :) 

An alternative could be repo for the pilot application that the user can push to. Something like `summit-2022-octo-pilot`

Comment (jimcadden) [reply .aicoe-ci.yaml:8]: Looking a couple steps ahead, we also need a home for the deployment manifests that GitOps/ACM will monitor. Maybe the `summit-2022-octo-pilot` repo could work for that as well? cc @cooktheryan

Comment (jimcadden) [reply .aicoe-ci.yaml:8]: > An alternative could be repo for the pilot application that the user can push to. Something like summit-2022-octo-pilot

Or, we keep using the existing repo, leave `aicoe-ci` unchanged, and we setup new pipelines to listen to the tag and build the car image. What do you guys think?

Comment (harshad16) [reply .aicoe-ci.yaml:8]: we can decide that based on the requirement. How many images do we need ?
from the above comments, it seems we need: docker car image, pilot notebook image ?

Comment (MichaelClifford) [reply .aicoe-ci.yaml:8]: Yes, one image to build the jupyter lab developer environment (like we've been doing). and one built from the docker file to deploy the model to the simulator or physical car.

Comment (harshad16) [reply .aicoe-ci.yaml:8]: ack, i will take my 1st statement back.
**meteor** doesn't rely on the `.aicoe-ci.yaml` file, so with this yaml file changing to dockerfile, we can still have this repo submitted to meteor and get a link for jupyterhub.
then when developer pushes a tag to this repo, we can build the image based on the specification described on the `.aicoe-ci.yaml` and  get a image in quay based on dockycar. 
which can be placed on the edge-device.

Comment (MichaelClifford) [reply .aicoe-ci.yaml:8]: In that case, I am good with the change :)

**Comment by harshad16** [comment]
Build have trouble building the dockerfile.
pulling the base image seems to be the major concern:
```
'STEP 1/5: FROM haoru233/tritonai-donkeycar:4.1-cpu
error creating build container: short-name resolution enforced but cannot prompt without a TTY'
```

File: .aicoe-ci.yaml (hunk 1 above)
Comment (jimcadden) [reply .aicoe-ci.yaml:8]: There remains the question about the "deploy" action. After the donkeycar container is built & pushed (presumably by aicoe-ci), we will need to run a custom task/pipeline to update the ACM config (via a commit) to begin the deployment. Is this something we can trigger via aicoe-ci?

**Comment by jimcadden** [comment]
> pulling the base image seems to be the major concern:

I've updated the base image url

File: .aicoe-ci.yaml (hunk 1 above)
Comment (harshad16) [reply .aicoe-ci.yaml:8]: we would have to create a custom task, lets discuss this in issue, so we can create the pipeline.__

File: apps/donkeycars/default_car/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,5 @@
+FROM docker.io/haoru233/tritonai-donkeycar:4.1-cpu
+ADD models/mypilot.h5 /donkeys/mycar/models/mypilot.h5
+ADD myconfig.py /donkeys/mycar/myconfig.py
```
Comment (harshad16) [inline apps/donkeycars/default_car/Dockerfile:3]: there is no `models` dir or `myconfig.py` in the repo, will this be added later?

**Review by harshad16 — CHANGES_REQUESTED** [review]
the build check is failing due to the falling issue

File: apps/donkeycars/default_car/Dockerfile (hunk 3 above)
Comment (jimcadden) [reply apps/donkeycars/default_car/Dockerfile:3]: > there is no models dir or myconfig.py in the repo, will this be added later?

These files exist in the repo already: https://github.com/AICoE/summit-2021-octo-keynote/blob/main/apps/donkeycars/default_car/models/

Problem here is that the paths in the Dockerfile are relative to its location in the src tree. Absolute paths should work, I think, e.g. `myconfig.py` -> `apps/donkeycars/default_car/myconfig.py`

Comment (harshad16) [reply apps/donkeycars/default_car/Dockerfile:3]: ah, i see, 
can we have them explicitly stated with the whole path 
or 
maybe we can use the `workdir` as the `apps/donkeycars/default_car`?

Comment (jimcadden) [reply apps/donkeycars/default_car/Dockerfile:3]: Via the Docker docs:
> paths are interpreted as relative to the source of the context of the build.

The context, in this case, appears to be the root directory of the repo

File: .aicoe-ci.yaml (hunk 1 above)
Comment (jimcadden) [reply .aicoe-ci.yaml:8]: > we would have to create a custom task, lets discuss this in issue, so we can create the pipeline.__

https://github.com/AICoE/summit-2021-octo-keynote/issues/44

**Review by harshad16 — APPROVED** [review]
lgtm
thanks :100:

**Pull request merged**